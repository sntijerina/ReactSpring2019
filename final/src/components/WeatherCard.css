.weather-card {
    display: flex;
    margin-top: 20px;
    width: 450px;
  }

  .weather-card-details {
    display:  flex;
    flex-direction: column;
  }

  .weather-card-image {
    display: flex;
    flex-direction: column;
    align-items: 'center';
    justify-content: center;
  }

  .weather-card-content {
    flex: '1 0 auto';
  }

  .weather-card-icon {
    width: 75px;
    height: 75px;
  }

  .weather-card-detail {
    margin-right: 5px;
  }