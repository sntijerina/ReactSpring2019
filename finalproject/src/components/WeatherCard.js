import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import CardMedia from "@material-ui/core/CardMedia";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";

import Clock from "./Clock";
import Weather from "./Weather";

const styles = theme => ({
  card: {
    maxWidth: 600,
    maxHeight: 400
  },
  actions: {
    display: "flex"
  }
});

class WeatherCard extends Component {
  render() {
    const { classes } = this.props;
    const { imgUrl } = this.props.icon;
    return (
            <Card 
              className={classes.card}
            >
            <CardHeader title={this.props.name} subheader={<Clock />} />
            <CardMedia>
              <img src={this.props.icon} alt={this.props.overcast} />
            </CardMedia>
            <CardContent>
              <Typography component="div" >
              <ul>
                <li>Current: <span>{this.props.temp}°</span></li>
                <li>High: <span>{this.props.maxTemp}°</span></li>
                <li>Low: <span>{this.props.minTemp}°</span></li>
              </ul>
              </Typography>
            </CardContent>
          </Card>
    );
  }
}

export default withStyles(styles)(WeatherCard);
