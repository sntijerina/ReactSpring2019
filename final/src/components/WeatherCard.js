import React, { Component } from "react";
import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import Typography from "@material-ui/core/Typography";

import './WeatherCard.css';

class WeatherCard extends Component {
  render() {

    return (
            <Card className="weather-card">
            <div className="weather-card-image">
              <img src={this.props.icon} alt={this.props.overcast} className="weather-card-icon"/>
            </div> 
            <div className="weather-card-details">
              <CardContent className="weather-card-content">
                <Typography component="h5" variant="h5" >
                  {this.props.name} {this.props.temp}°
                </Typography>
                <Typography variant="subtitle1">
                    {this.props.overcast}
                </Typography>
                <Typography variant="subtitle2">
                  <span className="weather-card-detail"><b>High:</b> {this.props.maxTemp}°</span>
                  <span className="weather-card-detail"><b>Low:</b> {this.props.minTemp}°</span>
                </Typography>
                <Typography variant="subtitle2">
                  <span className="weather-card-detail"><b>Humidity:</b> {this.props.humidity}%</span>
                  <span className="weather-card-detail"><b>Pressure:</b> {this.props.pressure}hPa</span>
                  <span className="weather-card-detail"><b>Wind:</b> {this.props.wind}m/h</span>
                </Typography>
              </CardContent>
            </div>
            
          </Card>
    );
  }
}

export default WeatherCard;