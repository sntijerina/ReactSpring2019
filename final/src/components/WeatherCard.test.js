import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import WeatherCard from './WeatherCard';
import Adapter from "enzyme-adapter-react-16";

Enzyme.configure({ adapter: new Adapter() });

describe('Weather Card', () => {

    it('Weather Card renders', () => {
        const component = shallow(<WeatherCard />);
    
        expect(component).toMatchSnapshot();
    });
});