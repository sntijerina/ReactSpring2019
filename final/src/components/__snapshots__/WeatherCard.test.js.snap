// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather Card Weather Card renders 1`] = `
<WithStyles(Card)
  className="weather-card"
>
  <div
    className="weather-card-image"
  >
    <img
      className="weather-card-icon"
    />
  </div>
  <div
    className="weather-card-details"
  >
    <WithStyles(CardContent)
      className="weather-card-content"
    >
      <WithStyles(Typography)
        component="h5"
        variant="h5"
      >
         
        °
      </WithStyles(Typography)>
      <WithStyles(Typography)
        variant="subtitle1"
      />
      <WithStyles(Typography)
        variant="subtitle2"
      >
        <span
          className="weather-card-detail"
        >
          <b>
            High:
          </b>
           
          °
        </span>
        <span
          className="weather-card-detail"
        >
          <b>
            Low:
          </b>
           
          °
        </span>
      </WithStyles(Typography)>
      <WithStyles(Typography)
        variant="subtitle2"
      >
        <span
          className="weather-card-detail"
        >
          <b>
            Humidity:
          </b>
           
          %
        </span>
        <span
          className="weather-card-detail"
        >
          <b>
            Pressure:
          </b>
           
          hPa
        </span>
        <span
          className="weather-card-detail"
        >
          <b>
            Wind:
          </b>
           
          m/h
        </span>
      </WithStyles(Typography)>
    </WithStyles(CardContent)>
  </div>
</WithStyles(Card)>
`;
